src/main/java/org/springframework/samples/petclinic/rest/advice/ExceptionControllerAdvice.java

```diff
@@ -46,29 +40,30 @@
 public class ExceptionControllerAdvice {
 
     /**
-     * Record for storing error information.
-     * <p>
-     * This record encapsulates the class name and message of the exception.
+     * Private method for constructing the {@link ProblemDetail} object passing the name and details of the exception class.
      *
-     * @param className The name of the exception class
-     * @param exMessage The message of the exception
+     * @param ex Object referring to the thrown exception.
+     * @param status HTTP response status.
      */
-    private record ErrorInfo(String className, String exMessage) {
-        public ErrorInfo(Exception ex) {
-            this(ex.getClass().getName(), ex.getLocalizedMessage());
-        }
```

Comment:
I suppose we have to rework the `openapi.yml` file in order to reference the problem details schema instead of the `RestError` one.

Reply:
Of course, I'll send the PR with the changes in the openapi.yml, along with some improvements to the ProblemDetail itself.